keps/sig-network/1435-mixed-protocol-lb/README.md

```diff
@@ -558,23 +557,29 @@ _This section must be completed when targeting beta graduation to a release._
 
 * **How can a rollout fail? Can it impact already running workloads?**
 
-  TBD
+  Enabling this feature gate moves the responsibility for handling mixed protocols from the Kubernetes API server to the specific CPI. See the provider-specific details at `API change and upgrade/downgrade situations`.
 
 * **What specific metrics should inform a rollback?**
 
-  TBD
+  As all providers either already have adjusted their error messages or intend to, enabling this feature gate may lead to a CPI error. In that case, remediation would be to disable the gate.
 
 * **Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?**
 
-  TBD
+This feature uses an existing value in existing fields, but with a different logic. If we create a Service with mixed protocol and then we roll back the API server to a version that does not implement this feature, the clients will still get the Service with mixed protocols when using the rolled-back API. If the client (CPI implementation) has been roll backed as well, then the client may encounter a Service setup that it does not support. All clouds either support this or adjusted their errors reported, per the enhancement issue.
+
 
 ### Monitoring Requirements
 
 _This section must be completed when targeting beta graduation to a release._
 
 * **How can an operator determine if the feature is in use by workloads?**
 
-  TBD
+Indirectly, via checking if a multi-protocol cloud load balancer is available.
```

Comment:
Specifically, that a service of `type: LoadBalancer` that uses two different protocols on the same port was validated and created by the API server?

Such an object could be created and persist in the DB after the feature flag is disabled. Can we confirm if we expect it to keep working in that circumstance, assuming no change to the CPI?

It would be handy if we were able to include any relevant metrics or log entries to be used to identify this in environments at scale.

Reply:
> Such an object could be created and persist in the DB after the feature flag is disabled. Can we confirm if we expect it to keep working in that circumstance, assuming no change to the CPI?

This is covered in detail in this section https://github.com/kubernetes/enhancements/blob/0c97a6dbe10d47252c72eee7466f5cbafdccf09e/keps/sig-network/1435-mixed-protocol-lb/README.md#api-change-and-upgradedowngrade-situations

Reply:
Let's link this here then.